src/HttpClient/HttpClient.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\HttpClient;
+
+use Sentry\Options;
+use Sentry\Util\Http;
+
+/**
+ * @internal
+ */
+class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkIdentifier;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkVersion;
+
+    public function __construct(string $sdkIdentifier, string $sdkVersion)
+    {
+        $this->sdkIdentifier = $sdkIdentifier;
+        $this->sdkVersion = $sdkVersion;
+    }
+
+    public function sendRequest(string $requestData, Options $options): Response
+    {
+        $dsn = $options->getDsn();
+        if (null === $dsn) {
+            throw new \RuntimeException('The DSN option must be set to use the HttpClient.');
+        }
+
+        $curlHandle = curl_init();
+        curl_setopt($curlHandle, \CURLOPT_URL, $dsn->getEnvelopeApiEndpointUrl());
+        curl_setopt($curlHandle, \CURLOPT_HTTPHEADER, Http::getRequestHeaders($dsn, $this->sdkIdentifier, $this->sdkVersion));
+        curl_setopt($curlHandle, \CURLOPT_USERAGENT, $this->sdkIdentifier . '/' . $this->sdkVersion);
+        curl_setopt($curlHandle, \CURLOPT_TIMEOUT, $options->getHttpTimeout());
+        curl_setopt($curlHandle, \CURLOPT_CONNECTTIMEOUT, $options->getHttpConnectTimeout());
+        curl_setopt($curlHandle, \CURLOPT_ENCODING, '');
+        curl_setopt($curlHandle, \CURLOPT_POST, true);
+        curl_setopt($curlHandle, \CURLOPT_POSTFIELDS, $requestData);
+        curl_setopt($curlHandle, \CURLOPT_RETURNTRANSFER, true);
```

Comment:
Since we don't actually use the body anywhere we should disable this to save on memory. Headers should still be returned (untested).

Reply:
> **CURLOPT_RETURNTRANSFER** `true` to return the transfer as a string of the return value of `curl_exec()` instead of outputting it directly.

If we remove this, we will just echo the response string. Better to keep it like that 😬

Reply:
Yikes! Good call 😉